src/main/java/org/fogbowcloud/arrebol/core/models/specification/Specification.java

```diff
@@ -3,69 +3,59 @@
+
+    private static final long serialVersionUID = 3435814833254660531L;
+
+    @Id
+    @GeneratedValue(strategy= GenerationType.IDENTITY)
+    String id;
     String cloudName;
     String image;
     String username;
     String privateKeyFilePath;
+
+    @Column(columnDefinition="varchar(512)")
     String publicKey;
     String contextScript;
-    String userDataFile;
-    String userDataType;
-
-    Map<String, String> requirements = new HashMap<String, String>();
+    //String userDataFile;
+    //String userDataType;
 
-    public Specification(String image, String username, String publicKey, String privateKeyFilePath) {
-        this(image, username, publicKey, "", privateKeyFilePath, "", "");
-    }
+    @ElementCollection
+    Map<String, String> requirements;
 
-    public Specification(String image, String username, String publicKey, String privateKeyFilePath, String cloudName){
-        this(image, username, publicKey, privateKeyFilePath, cloudName, "", "");
+    public Specification(String image, String username, String publicKey, String privateKeyFilePath, Map<String, String> requirements) {
+        this(image, username, publicKey, privateKeyFilePath, requirements, null);
     }
 
-    public Specification(String image, String username, String publicKey, String privateKeyFilePath, String cloudName,
-                         String userDataFile, String userDataType) {
+    public Specification(String image, String username, String publicKey, String privateKeyFilePath, Map<String, String> requirements, String cloudName) {
         this.cloudName = cloudName;
         this.image = image;
         this.username = username;
         this.publicKey = publicKey;
         this.privateKeyFilePath = privateKeyFilePath;
-        this.userDataFile = userDataFile;
-        this.userDataType = userDataType;
+        this.requirements = requirements;
     }
 
-    public Specification(){}
-
-    public void addRequirement(String key, String value) {
-        requirements.put(key, value);
+    Specification(){
+        //Default constructor
     }
 
     public String getRequirementValue(String key) {
         return requirements.get(key);
     }
 
-    public void putAllRequirements(Map<String, String> requirements) {
-        for (Map.Entry<String, String> e : requirements.entrySet()) {
-
-            this.requirements.put(e.getKey(), e.getValue());
-        }
-    }
-
-    public Map<String, String> getAllRequirements() {
+    public Map<String, String> getRequirements() {
         return requirements;
     }
 
-    public void removeAllRequirements() {
-        requirements = new HashMap<String, String>();
-    }
-
-    public boolean parseToJsonFile(String jsonDestFilePath) {
-
-        List<Specification> spec = new ArrayList<Specification>();
-        spec.add(this);
-        return SpecificationUtils.parseSpecsToJsonFile(spec, jsonDestFilePath);
+    public void setRequirements(Map<String, String> requirements){
```

Comment:
why do we need this set method? it seems to me a Spec should not be modified after creation

Reply:
Removed. It was unnecessary.